### ISSUE TITLE
Is there a development roadmap visible to the public?

### ISSUE THREAD

**Issue by Ichaelus** [jruby/jruby#4963]
Hi there,

this is not a bug per se, but I was unable to find a development roadmap or further information about the near and midterm future of this project. 

The [contribution page](http://jruby.org/contribute) refers to development to bug-fixing, but what about feature development?

Thanks for helping me out and keep on coding,
Michael

**Comment by enebo** [comment]
There is not a concrete roadmap document.  Typically during conferences we talk about intentions but those intentions are often wrong.  We planned on JRuby 9.2.x around the new year (well we had planned on releasing it early in 2017 originally at one point too) for Ruby 2.4 support but that likely will be Februaryish.  2.5 support 9.3.x is being worked on but we have no date we are even thinking about yet.  Our 2.3 support is 9.1.x and we have been getting pretty close to a release a month the last few point releases.

This is somewhat unsatisfying and we could make a document but people also get unsatisfied when we miss every date on it.  A mixed bag.  We have been concerned recently that we appear to be over a year behind MRI now.  We attribute much of this to needing to work on supplementary projects like activerecord-jdbc.  That works is almost caught up though.  So we hope the 2.5 turn around will be much less than a year.

**Comment by Ichaelus** [comment]
Thank you @enebo for giving me some insight!
As it is hard to maintain a document and stick to its promises, a roadmap such as [the one of MRI](https://bugs.ruby-lang.org/projects/ruby-trunk/roadmap) (without due dates) might help outstanding people such as me keeping track of the projects state.

Besides from keeping up compatibility with MRI and resolving issues, are there plans or milestones for other features? (E.g. performance-based)

Thanks again

**Comment by enebo** [comment]
@Ichaelus Interestingly, we do have issues for 2.4 compat and 2.5 compat.  I wonder if we can make a label which is 'feature' so we can just generate a list of all 'features' we are or will be working on.  Some of those will have milestones on them so it will partially address your original question too.  We just need to make it obvious people can look.

 We have various plans here and there that we know about but we could outline them as feature issues.  As far as when those land they typically take a backseat to compatibility, but we should still outline activities like: profiler+inline, bytelisting all the things, redesign constant access instrs, ...

I will add this and see what we can do to make this easier for people to find.

**Comment by headius** [comment]
Yeah I want to echo @enebo here...our roadmap is in many ways embedded in the issue tracker, and to some extent in the wiki. We don't set dates because the types of issues and enhancements we work on often are of unknown difficulty.

We do try to associate issues with an upcoming release we intend for them to go into, which is what that MRI link appears to be. And issues that are noisy get attention more quickly :-)
